src/main/scala/barstools/tapeout/transforms/GenerateTopAndHarness.scala

```diff
@@ -7,118 +7,78 @@ import firrtl.ir._
 import firrtl.options.{Dependency, InputAnnotationFileAnnotation, StageMain}
 import firrtl.passes.memlib.ReplSeqMemAnnotation
 import firrtl.stage.{FirrtlCircuitAnnotation, FirrtlStage, OutputFileAnnotation, RunFirrtlTransformAnnotation}
+import firrtl.passes.{ConvertFixedToSInt}
 import firrtl.transforms.BlackBoxResourceFileNameAnno
 import logger.LazyLogging
 
 // Requires two phases, one to collect modules below synTop in the hierarchy
 // and a second to remove those modules to generate the test harness
 private class GenerateTopAndHarness(annotations: AnnotationSeq) extends LazyLogging {
-  val synTop:         Option[String] = annotations.collectFirst { case SynTopAnnotation(s) => s }
-  val topFir:         Option[String] = annotations.collectFirst { case TopFirAnnotation(s) => s }
-  val harnessFir:     Option[String] = annotations.collectFirst { case HarnessFirAnnotation(s) => s }
-  val topAnnoOut:     Option[String] = annotations.collectFirst { case TopAnnoOutAnnotation(s) => s }
-  val harnessAnnoOut: Option[String] = annotations.collectFirst { case HarnessAnnoOutAnnotation(s) => s }
-  val harnessTop:     Option[String] = annotations.collectFirst { case HarnessTopAnnotation(h) => h }
-  val harnessConf:    Option[String] = annotations.collectFirst { case HarnessConfAnnotation(h) => h }
-  val harnessOutput:  Option[String] = annotations.collectFirst { case HarnessOutputAnnotation(h) => h }
-  val topDotfOut:     Option[String] = annotations.collectFirst { case TopDotfOutAnnotation(h) => h }
-  val harnessDotfOut: Option[String] = annotations.collectFirst { case HarnessDotfOutAnnotation(h) => h }
+  val outAnno:     Option[String] = annotations.collectFirst { case OutAnnoAnnotation(s) => s }
+  val harnessConf: Option[String] = annotations.collectFirst { case HarnessConfAnnotation(h) => h }
 
   val annoFiles: List[String] = annotations.flatMap {
     case InputAnnotationFileAnnotation(f) => Some(f)
     case _                                => None
   }.toList
 
-  lazy val rootCircuitTarget = CircuitTarget(harnessTop.get)
-
-  val topAnnos = synTop.map(st => ReParentCircuitAnnotation(rootCircuitTarget.module(st))) ++
-    topDotfOut.map(BlackBoxResourceFileNameAnno)
-
   // Dump firrtl and annotation files
   protected def dump(
     circuit:     Circuit,
-    annotations: AnnotationSeq,
-    firFile:     Option[String],
-    annoFile:    Option[String]
+    annotations: AnnotationSeq
```

Comment:
I would rename this function as `dumpAnnos` and remove the `circuit` parameter.

Reply:
I also generally don't like functions that read global parameters (instead of using function parameters). For now this is fine, but could you add a comment that this function reads the global `outAnno`.